nuxeo-common/src/main/java/org/nuxeo/common/utils/PeriodAndDuration.java

```diff
@@ -0,0 +1,228 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *       Florent Guillaume
+ */
+package org.nuxeo.common.utils;
+
+import java.time.Duration;
+import java.time.Period;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * The combination of a {@link Period} and a {@link Duration}.
+ * <p>
+ * This allows the representation of ISO 8601 "durations", which comprise a nominal duration (Java {@link Period}, i.e.,
+ * years, months, days), and an accurate duration (Java {@link Duration}, i.e., hours, minutes, seconds).
+ *
+ * @since 11.1
+ */
+public final class PeriodAndDuration {
+
+    /**
+     * A constant for a period and duration of zero.
+     */
+    public static final PeriodAndDuration ZERO = new PeriodAndDuration(Period.ZERO, Duration.ZERO);
+
+    protected static final Pattern PATTERN = Pattern.compile("([-+]?)P" //
+            + "(?:([-+]?[0-9]+)Y)?" //
+            + "(?:([-+]?[0-9]+)M)?" //
+            + "(?:([-+]?[0-9]+)D)?" //
+            + "(T" //
+            + "(?:([-+]?[0-9]+)H)?" //
+            + "(?:([-+]?[0-9]+)M)?" //
+            + "(?:([-+]?[0-9]+)(?:[.,]([0-9]{0,9}))?S)?" //
+            + ")?", //
+            Pattern.CASE_INSENSITIVE);
+
+    protected static final int SECONDS_PER_MINUTE = 60;
+
+    protected static final int SECONDS_PER_HOUR = 60 * SECONDS_PER_MINUTE;
+
+    public final Period period;
+
+    public final Duration duration;
+
+    /**
+     * Constructs a {@link PeriodAndDuration} from the given period and duration.
+     */
+    public PeriodAndDuration(Period period, Duration duration) {
+        Objects.requireNonNull(period, "period");
+        Objects.requireNonNull(duration, "duration");
+        this.period = period;
+        this.duration = duration;
+    }
+
+    /**
+     * Constructs a {@link PeriodAndDuration} from the given period.
+     */
+    public PeriodAndDuration(Period period) {
+        this(period, Duration.ZERO);
+    }
+
+    /**
+     * Constructs a {@link PeriodAndDuration} from the given duration.
+     */
+    public PeriodAndDuration(Duration duration) {
+        this(Period.ZERO, duration);
+    }
+
+    /**
+     * Adds this {@link PeriodAndDuration} to the given date/time and returns the result.
+     *
+     * @param dateTime the date/time to add
+     * @return the date/time to which the period and duration have been added
+     */
+    public ZonedDateTime plus(ZonedDateTime dateTime) {
+        return dateTime.plus(period).plus(duration);
```

Comment:
Could be interesting to make this class implement `TemporalAmount`, as it we could use `ZonedDateTime#plus(PeriodAndDuration)`, but it requires more method to implement (4 methods).
Otherwise, it seems that `addTo` is preferred on a temporal amount. Wdyt?

Reply:
I initially thought `TemporalAmount` would be too complex to implement but indeed on second look it doesn't seem to be the case, so I'll take a stab. Noted for `addTo` as well in case I don't do it.

Reply:
Done, please re-review